psalm.xml

```
@@ -1,46 +1,81 @@
-        <directory name="vendor-bin/psalm/vendor" />
-        <ignoreFiles>
-            <directory name="vendor/bin" />
-        </ignoreFiles>
-    </extraFiles>
-    <issueHandlers>
-        <UndefinedClass>
-            <errorLevel type="suppress">
-                <referencedClass name="Doctrine\DBAL\Types\Type" />
+	<stubs>
+		<file name="tests/stub.phpstub" preloadClasses="true"/>
+	</stubs>
+	<projectFiles>
+		<directory name="lib" />
+		<ignoreFiles>
+			<directory name="vendor" />
+			<directory name="vendor-bin" />
+		</ignoreFiles>
+	</projectFiles>
+	<extraFiles>
+		<directory name="vendor" />
+		<directory name="vendor-bin/psalm/vendor" />
+		<ignoreFiles>
+			<directory name="vendor/bin" />
+		</ignoreFiles>
+	</extraFiles>
+	<issueHandlers>
+		<UndefinedClass>
+			<errorLevel type="suppress">
+				<referencedClass name="Doctrine\DBAL\Types\Type" />
 				<referencedClass name="OCP\SetupCheck\ISetupCheck" />
-            </errorLevel>
-        </UndefinedClass>
-        <UndefinedDocblockClass>
-            <errorLevel type="suppress">
-                <referencedClass name="Doctrine\DBAL\Driver\Statement" />
-                <referencedClass name="Doctrine\DBAL\Schema\Schema" />
-                <referencedClass name="Doctrine\DBAL\Schema\SchemaException" />
-                <referencedClass name="Doctrine\DBAL\Schema\Table" />
-            </errorLevel>
-        </UndefinedDocblockClass>
-    </issueHandlers>
+			</errorLevel>
+		</UndefinedClass>
+		<UndefinedDocblockClass>
+			<errorLevel type="suppress">
+				<referencedClass name="Doctrine\DBAL\Driver\Statement" />
+				<referencedClass name="Doctrine\DBAL\Schema\Schema" />
+				<referencedClass name="Doctrine\DBAL\Schema\SchemaException" />
+				<referencedClass name="Doctrine\DBAL\Schema\Table" />
+			</errorLevel>
+		</UndefinedDocblockClass>
+		<DeprecatedClass>
+			<errorLevel type="suppress">
+				<directory name="lib" />
+			</errorLevel>
+		</DeprecatedClass>
```

I'm not quite sure I understand what this does...

`type="suppress"` sounds like it suppresses the errors rather than complaining.


Since the commit message sounds like it treats more things as errors than before I assume that just mentioning these issue handlers activates them and they will only be suppressed in the `lib` directory. But then again... isn't most php code in lib? This seems to stem from server where there's also `core`, `aps`, `ocs`... so there it makes sense to me. But in collectives? Do we actually need to exclude `lib`?

---

Indeed, the changes were wrong. I started working on bumping the psalm error level now instead. This PR bumps the error level to `4` and brings many fixed reported by error level `3`. There's still around 57 errors to be fixed though, which I'd tackle later on in a follow-up PR.